ignite/pkg/jsonfile/jsonfile.go

```diff
@@ -0,0 +1,311 @@
+package jsonfile
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/pkg/errors"
+
+	"github.com/ignite-hq/cli/ignite/pkg/tarball"
+)
+
+const (
+	keySeparator = "."
+)
+
+type (
+	// JSONFile represents the file
+	JSONFile struct {
+		file        readWriteSeeker
+		tarballPath string
+		url         string
+		updates     map[string][]byte
+		cache       []byte
+	}
+	// UpdateFileOption configures file update.
+	UpdateFileOption func(map[string][]byte)
+
+	writeTruncate interface {
+		Truncate(size int64) error
+	}
+	readWriteSeeker interface {
+		io.ReadWriteSeeker
+		Close() error
+		Sync() error
+	}
+)
+
+var (
```

Comment:
Let's move this block to the begginnig of the file and put newlines between errors.